Assets/Features/Building/Scripts/Grid/GridPosition.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using UnityEngine;
+
+namespace Features.Building.Scripts.Grid
+{
+    [Serializable]
+    public class GridPosition
+    {
+        public GridPosition(Vector2Int position, GridLayer layer)
+        {
+            Position = position;
+            Layer = layer;
+        }
+
+        public Vector2Int Position;
```

Comment:
Seems like you don't want other classes to be able to change these two values. Make them properties instead. Plus if all content is public, consider making it a struct

Reply:
Changed this to a private Tile with public Tile Getter. With public tile {get; private set} it won't be shown in the editor